glusterfs/server/setup.sls

```diff
@@ -58,14 +58,18 @@ glusterfs_vol_{{ name }}:
 {%- endif %}
 
 glusterfs_vol_{{ name }}_start:
+  {%- if force_compatibility %}
+  cmd.run:
+    - name: |
+        gluster volume info {{ name }} | grep "Status: Started" ||  gluster volume start {{ name }}
```

Comment:
You should use onlyif or unless parameter for check if it's required to start the volume.